google/resource_container_cluster.go

```diff
@@ -698,6 +699,31 @@ func resourceNodeConfigEmptyGuestAccelerator(diff *schema.ResourceDiff, meta int
 	return nil
 }
 
+func resourceContainerClusterNodeVersionCustomizeDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	// separate func to allow unit testing
+	return resourceContainerClusterNodeVersionCustomizeDiffFunc(diff)
+}
+
+// resourceContainerClusterNodeVersionCustomizeDiffFunc is checking that argument node_version have
+// same value as min_master_version if setted on create.
+func resourceContainerClusterNodeVersionCustomizeDiffFunc(diff TerraformResourceDiff) error {
+
+	// By pass the diff if we are not on create.
+	oldValueName, _ := diff.GetChange("name")
+	if oldValueName != "" {
+		return nil
+	}
+
+	_, newValueNode := diff.GetChange("node_version")
+	_, newValueMaster := diff.GetChange("min_master_version")
+
+	if newValueNode != "" && newValueNode != newValueMaster {
```

Comment:
Can you check if `newValueMaster` is `""` as well? Terraform has a concept of an "unknown" value that's only discovered later in plan that will appear here as `""`. It's unlikely a node_version will be known while min_master_version isn't, but adding a guard is easy enough.